src/go/rpk/pkg/config/config.go

```diff
@@ -17,12 +17,13 @@ import (
 )
 
 const (
-	DefaultKafkaPort     = 9092
-	DefaultSchemaRegPort = 8081
-	DefaultProxyPort     = 8082
-	DefaultAdminPort     = 9644
-	DefaultRPCPort       = 33145
-	DefaultListenAddress = "0.0.0.0"
+	DefaultKafkaPort         = 9092
+	DefaultSchemaRegPort     = 8081
+	DefaultProxyPort         = 8082
+	DefaultAdminPort         = 9644
+	DefaultRPCPort           = 33145
+	DefaultListenAddress     = "0.0.0.0"
+	DefaultAdvertisedAddress = "127.0.0.1"
```

Comment:
Nit: can this be named something like Localhost or LoopbackIP?

The other default variables are defaults that _we_ add internally (in params.go). This one here is only a default _externally_, in the repo level redpanda.yaml file. I was confused / looking up where this default is coming from -- it isn't rpk, rpk isn't defaulting this. The PR is to guard against behavior that may be set externally.

Reply:
Actually I may be wrong. Where is this being set to 127.0.0.1? I thought that example file was only used for testing.

Reply:
Yes, it is used for testing. But for some Redpanda images this is the starting `redpanda.yaml`. 

But yeah, I agree. We can change the name to `localhost` but I wanted to make sure that I was using `default` as in "this comes from somewhere else" and not just only: "we will replace localhost".

Reply:
Approved but lmk if you change the name and I'll re-+1.

Reply:
I renamed the variable here: https://github.com/redpanda-data/redpanda/compare/a86992b5e41e313d2241026dabe5ae58e7cf77ca..cd0faa8853477ea3ba64053f4b18a1f90d4a538f